src/core/Akka.Persistence/Serialization/PersistenceMessageSerializer.cs

```diff
@@ -0,0 +1,167 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="IMessage.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Akka.Actor;
+using Akka.Persistence.Serialization.Proto.Msg;
+using Akka.Serialization;
+using Google.Protobuf;
+
+namespace Akka.Persistence.Serialization
+{
+    public class PersistenceMessageSerializer : Serializer
+    {
+        public PersistenceMessageSerializer(ExtendedActorSystem system) : base(system)
+        {
+            IncludeManifest = true;
+        }
+
+        public override bool IncludeManifest { get; }
+
+        public override byte[] ToBinary(object obj)
+        {
+            if (obj is IPersistentRepresentation) return GetPersistentMessage((IPersistentRepresentation)obj).ToByteArray();
+            if (obj is AtomicWrite) return GetAtomicWrite((AtomicWrite)obj).ToByteArray();
+            if (obj is AtLeastOnceDeliverySnapshot) return GetAtLeastOnceDeliverySnapshot((AtLeastOnceDeliverySnapshot)obj).ToByteArray();
+
+            throw new ArgumentException($"Can't serialize object of type [{obj.GetType()}] in [{GetType()}]");
+        }
+
+        private PersistentMessage GetPersistentMessage(IPersistentRepresentation persistent)
+        {
+            PersistentMessage message = new PersistentMessage();
+
+            if (persistent.PersistenceId != null) message.PersistenceId = persistent.PersistenceId;
+            if (persistent.Manifest != null) message.Manifest = persistent.Manifest;
+            if (persistent.WriterGuid != null) message.WriterGuid = persistent.WriterGuid;
+            if (persistent.Sender != null) message.Sender = Akka.Serialization.Serialization.SerializedActorPath(persistent.Sender);
+
+            message.Payload = GetPersistentPayload(persistent.Payload);
+            message.SequenceNr = persistent.SequenceNr;
+            message.Deleted = persistent.IsDeleted;
+
+            return message;
+        }
+
+        private PersistentPayload GetPersistentPayload(object obj)
+        {
+            Serializer serializer = system.Serialization.FindSerializerFor(obj);
```

Comment:
Needs DefaultSerializer persistence setting support.

Reply:
But i dont consider it a blocking issue. We can fix it later. Its not a thing that will break any existing stuff

Reply:
As we talked about in chat, the current dev branch does not support specifying the default serializer for persistence serialization, it just uses whatever has been defined for the objects in the serialization-bindings.

Currently the dev branch doesn't actually support custom serializers at all for the payload within the IPersistentRepresentation, this was fixed and is reflected in the unit test changes.